### PR TITLE
VMware: Add basic vmware module related guide

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
This fix adds page which tries to consolidate various information related
to vmware modules and debugging related information.

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME

docs/docsite/rst/guide_vmware.rst
docs/docsite/rst/guides.rst

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

File: docs/docsite/rst/guide_vmware.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+Ansible provides various modules to manage VMware infrastructure which includes datacenter, cluster,
+host system and virtual machine etc.
+
+Requirements
+````````````
+
+Ansible VMware modules are written on top of `pyVmomi <https://github.com/vmware/pyvmomi>`_.
+pyVmomi is the Python SDK for the VMware vSphere API that allows user to manage ESX, ESXi,
+and vCenter infrastcture. User can install pyVmomi using
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst]: infrastcture/infrastructure/

File: docs/docsite/rst/guide_vmware.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
+Getting Started with VMware
+===========================
+
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst:3]: Please add table of contents with:
`.. contents:: Topics`

File: docs/docsite/rst/guide_vmware.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Introduction
+````````````
+
+Ansible provides various modules to manage VMware infrastructure which includes datacenter, cluster,
+host system and virtual machine etc.
+
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst]: Link to VMware module list?

File: docs/docsite/rst/guide_vmware.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+.. code-block:: bash
+
+    $ pip install pyvmomi
+
+
+vmware_guest module
+```````````````````
+
+The ``vmware_guest`` module is used to manage various operations related to virtual machines in
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst]: `The :ref:`vmware_guest <vmware_guest>` module...`

File: docs/docsite/rst/guide_vmware.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+From Ansible version 2.5 and onwards, ``folder`` is still an optional parameter with no default value.
+This parameter will be now used to identify user desired virtual machine, if multiple virtual machines or virtual
+machine templates are found with same name. VMware does not restrict system administrator from creating virtual
+machine with same name.
+
+Debugging
+`````````
+
+In order to understand the issue related to VMware modules, one must understand infrastucture details. So, while
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst]: infrastructure

File: docs/docsite/rst/guide_vmware.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+
+.. code-block:: bash
+
+    $ export GOVC_USERNAME=<ESXI_OR_VCENTER_USERNAME>
+    $ export GOVC_PASSWORD=<ESXI_OR_VCENTER_PASSWORD>
+    $ export GOVC_URL=https://<ESXI_OR_VCENTER_HOSTNAME>:443
+    $ govc find /
+
+
```
Comment (gundalow) [inline docs/docsite/rst/guide_vmware.rst]: Do you want to mention the VMware working group?
github.com/ansible/community/tree/master/group-vmware

**Review by gundalow — CHANGES_REQUESTED** [review]

File: docs/docsite/rst/guide_vmware.rst (hunk 2 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst:3]: OK.

File: docs/docsite/rst/guide_vmware.rst (hunk 3 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst]: OK.

File: docs/docsite/rst/guide_vmware.rst (hunk 1 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst]: OK.

File: docs/docsite/rst/guide_vmware.rst (hunk 4 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst]: ok.

File: docs/docsite/rst/guide_vmware.rst (hunk 5 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst]: OK.

File: docs/docsite/rst/guide_vmware.rst (hunk 6 above)
Comment (Akasurde) [reply docs/docsite/rst/guide_vmware.rst]: I didn't see any references of working group in other guides, so I didn't add it here.

File: docs/docsite/rst/guide_vmware.rst (hunk 2 above)
Comment (dharmabumstead) [reply docs/docsite/rst/guide_vmware.rst:3]: No, we don't need a TOC here.

**Review by dharmabumstead — APPROVED** [review]
OK with some edits.

**Review by gundalow — APPROVED** [review]

**Pull request merged**